### ISSUE TITLE
regexp loading false error

### ISSUE THREAD

**Issue by Mantou1233** [oven-sh/bun#1486]
### What version of Bun is running?

0.2.2

### What platform is your computer?

Linux 4.4.0-19041-Microsoft #1237-Microsoft Sat Sep 11 14:32:00 PST 2021 x86_64 x86_64

### What steps can reproduce the bug?

install discord.js
cannot load because 
![image](https://user-images.githubusercontent.com/74487898/201357921-b8a226a6-1720-4eac-b24a-5cfa59455bed.png)
regexp cannot load

### How often does it reproduce? Is there a required condition?

n

### What is the expected behavior?

it loading manually

### What do you see instead?

![image](https://user-images.githubusercontent.com/74487898/201357586-c684a50e-4aa2-40de-bfff-5b843b3d233a.png)

### Additional information

n

**Comment by Jarred-Sumner** [comment]
cc @dylan-conway

**Comment by dylan-conway** [comment]
@Jarred-Sumner oniguruma requires only the property value in property escapes. The fix would be to remove the equals sign and almost everything before it (`\p{^Script=Deva}` => `\p{^Deva}` and `\p{Script=Thai}` => `\p{Thai}`). The problem is there doesn't seem to be support for Script_Extensions:
```js
"٢".match(/\p{Script=Thaana}/u);
// null as Thaana is not the predominant script

"٢".match(/\p{Script_Extensions=Thaana}/u);
// ["٢", index: 0, input: "٢", groups: undefined]
```
Both will return null with oniguruma. I'm pretty sure discord.js doesn’t use Script_Extensions anywhere so it’s not a problem here, but other libraries might

**Comment by Jarred-Sumner** [comment]
Fixed in Bun 0.3.1
